src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -277,8 +287,12 @@ bool LibraryCallKit::arch_supports_vector(int sopc, int num_elem, BasicType type
   }
 
   if ((mask_use_type & VecMaskUsePred) != 0) {
-    if (!Matcher::has_predicated_vectors() ||
-        !Matcher::match_rule_supported_vector_masked(sopc, num_elem, type)) {
+    if (!Matcher::has_predicated_vectors()) {
+      return false;
```

Comment:
If we return here, we would miss the intrinsic failing msg "Rejected vector mask predicate using ...", right?